tsl/src/continuous_aggs/create.c

```diff
@@ -1334,6 +1346,24 @@ mattablecolumninfo_addentry(MatTableColumnInfo *out, Node *input, int original_q
 			}
 		}
 		break;
+		case T_Var:
+		{
+			PRINT_MATCOLNAME(colbuf, "var", original_query_resno, matcolno);
```

Comment:
We should maintain 2 invariants:
1. every aggref gets its own column in the materialized hypertable
2. every grouby element gets its own column in the materialized hypertable.

Prior to this PR, the code adds a column for any target entry exprs like var1+var2 as well. We do not need this additioanl column in the materialized hypertable (Note due SQL grammar rules var1 and var2 are group by columns or var1+var2 is a group by column and already exists in the materialized hypertable). But we do not want to modify exprs of this form ( that involve only vars) as it would break compatibility with the existing materialized hypertable defn. (We should do this eventually and provide an upgrade path for materialized hypertable schema changes).

The current code has the following problem with exprs that involve a combination of aggrefs + grouping columns: 
SELECT  expr( aggref, group by column) -----> problematic expr

The issue is : the target entry is not fixed up for vars, the group-by column reference (i.e. does not refer to the correct column in the materialized hypertable) . The expr is fixed up only for the aggrefs. The correct solution is to fix up the var references to point to the correct attnos from the materialized hypertable instead of ADDING a new column for it.
This will ensure that we are compatibile with existing materialized hypertables and will allow us to fix up the finalize view definition for caggs with this issue(instead of having to drop and add the cagg).
e.g.
create materialized view c4 with (timescaledb.continuous) as select time_bucket( '1 day' ::interval, "Time") bkt, sum(b) + (i +b) + b from test1 group by  bkt, b, (i+b) ;

Reply:
The motivation behind @gayyappan 's comment is the potential to do data recovery during an update (or post-update with some tools) on the materialized table data, since the existing issue leads to crashes or data loss/corruption.

The current state of this PR does not repair any preexisting data, it only makes sure that the new CAGG creations do not have the issue.